src/cpp/core/FileSerializer.cpp

```diff
@@ -125,8 +125,8 @@ Error writeStringToFile(const FilePath& filePath,
    using namespace boost::system::errc ;
    
    // open file
-   boost::shared_ptr<std::ostream> pOfs;
-   Error error = filePath.open_w(&pOfs, truncate);
+   std::shared_ptr<std::ostream> pOfs;
```

Comment:
Sorry if this has already been discussed. Do we plan on making the switch to always using `std::shared_ptr<>`? Or are we going to live with a mix of the two in the codebase?

Reply:
(Note: not proposing we do this for this PR; rather, that we consider doing it for later)

Reply:
I think it would be a good idea to always use `std::shared_ptr<>` in the future, but like you said, it would be something to consider doing in a separate PR.